snuba/datasets/storage.py

```diff
@@ -71,6 +72,16 @@ def get_query_processors(self) -> Sequence[QueryProcessor]:
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def get_query_splitters(self) -> Sequence[QuerySplitStrategy]:
+        """
+        If this storage supports splitting queries as optimizations, they are provided here.
+        These are optimizations, the query plan builder may decide to override the storage
+        and to skip the splitters. So correctness of the query must not depend on these
+        strategies to be applied.
+        """
+        raise NotImplementedError
```

Comment:
I think it makes sense in general. Will add. Though I think it will have very little impact concerning joins because joinedStorage is supposed to die as soon as I have time to revamp this: https://github.com/getsentry/snuba/pull/830

And actually have a join query joining two storages instead of having a fake single storage that could never be plugged into an entity anyway.